src/GovUk.Education.ExploreEducationStatistics.Data.Model/Repository/FootnoteRepository.cs

```diff
@@ -19,40 +18,134 @@ public FootnoteRepository(StatisticsDbContext context)
+                FilterItems = filterItemIds.Select(filterItemId => new FilterItemFootnote
+                {
+                    FilterItemId = filterItemId
+                }).ToList(),
+                Indicators = indicatorIds.Select(indicatorId => new IndicatorFootnote
+                {
+                    IndicatorId = indicatorId
+                }).ToList(),
+                Subjects = subjectIds.Select(subjectId => new SubjectFootnote
+                {
+                    SubjectId = subjectId
+                }).ToList()
+            };
+
+            // TODO EES-2971 We could link Footnote with ReleaseSubject to simplify getting all foonotes for a subject.
+            // Currently this involves getting all footnotes with criteria matching the subject directly or indirectly.
+            // This would also prevent removing a subject from a release until all of its footnote links are removed.
+            // Currently a footnote could remain linked to a release and a subject (either directly or indirectly via
+            // its FilterItems/FilterGroups/Filters/Indicators) even though no ReleaseSubject exists.
+
+            await _context.Footnote.AddAsync(footnote);
+            await _context.SaveChangesAsync();
+
+            return footnote;
+        }
+
+        public async Task<List<Footnote>> GetFilteredFootnotes(
             Guid releaseId,
             Guid subjectId,
             IEnumerable<Guid> filterItemIds,
             IEnumerable<Guid> indicatorIds)
         {
-            var releaseIdParam = new SqlParameter("releaseId", releaseId);
-            var subjectIdParam = new SqlParameter("subjectId", subjectId);
-            var filterItemListParam = CreateIdListType("filterItemList", filterItemIds);
-            var indicatorListParam = CreateIdListType("indicatorList", indicatorIds);
+            // List the id's of all filter items, filter groups, filters, indicators
+            // Filter out any of these which don't belong to the same subject
+            var filterItemsQueryable = _context.FilterItem
+                .Where(filterItem => filterItemIds.Contains(filterItem.Id)
+                                     && filterItem.FilterGroup.Filter.SubjectId == subjectId);
+
+            var filterItems = await filterItemsQueryable
+                .Select(filterItem => filterItem.Id)
+                .ToListAsync();
+
+            var filterGroups = await filterItemsQueryable
+                .Select(filterItem => filterItem.FilterGroupId)
+                .Distinct()
+                .ToListAsync();
+
+            var filters = await filterItemsQueryable
+                .Select(filterItem => filterItem.FilterGroup.FilterId)
+                .Distinct()
+                .ToListAsync();
+
+            var indicators = await _context.Indicator
+                .Where(indicator => indicatorIds.Contains(indicator.Id)
+                                    && indicator.IndicatorGroup.SubjectId == subjectId)
```

Comment:
Ah yeah, good spot! I'm surprised we've never seen this but I suppose it's because the table tool always requires at least one indicator to be selected. I've made this change in `ResultSubjectMetaService` on a new commit, thanks.